module/VuFind/src/VuFind/Db/Service/AbstractDbService.php

```diff
@@ -82,6 +85,11 @@ protected function getEntityClass(string $entity): string
      */
     public function persistEntity(EntityInterface $entity): void
     {
+        $config = $this->entityManager->getConfiguration();
```

Comment:
This is not the right configuration, unfortunately. You need the top-level VuFind configuration. I think the best way to make that available would be to add a third constructor parameter of type `VuFind\Config\Config` and then pass in the value through AbstractDbServiceFactory. You can retrieve the appropriate config in the factory using: `$container->get(\VuFind\Config\PluginManager::class)->get('config')`.

Reply:
(Another option would be to add a `bool $privacyMode` constructor parameter, and read the configuration in the factory to pass just the one relevant setting... depends on whether or not we want broad access to the configuration in this class).